elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala

```diff
@@ -128,8 +128,18 @@ class ElasticSearchQueryProcessor(val logicalPlan: LogicalPlan, val parameters:
   }
 
   private def selectFields(fields: Seq[Attribute], query: SearchDefinition): SearchDefinition = {
-      val stringFields: Seq[String] = fields.map(_.name)
-      query.fields(stringFields.toList: _*)
+      val subDocuments = schemaProvided.toSeq flatMap {
+        _.fields collect {
+          case StructField(name, _: StructType, _, _) => name
+        }
+      }
+      val stringFields: Seq[String] = fields.view map (_.name) filterNot (subDocuments contains _)
+
+      val fieldsQuery = query.fields(stringFields.toList: _*)
+
+      if(stringFields.size != fields.size)
```

Comment:
Maybe it could be more precise and elegant to use: 

`stringFields.containsAll(fields)`